Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIPermissionsCocoa.mm

```diff
@@ -99,12 +162,23 @@
     completionHandler(true);
 }
 
-void WebExtensionContext::parseMatchPatterns(HashSet<String> origins, HashSet<Ref<WebExtensionMatchPattern>>& matchPatterns)
+void WebExtensionContext::firePermissionsEventListenerIfNecessary(NSString *notificationName, const PermissionsSet& permissions, const MatchPatternSet& matchPatterns)
 {
-    for (auto& origin : origins) {
-        auto pattern = WebExtensionMatchPattern::getOrCreate(static_cast<NSString *>(origin));
-        matchPatterns.add(*pattern);
-    }
+    WebExtensionEventListenerType type;
+    if ([notificationName isEqualToString:_WKWebExtensionContextPermissionsWereGrantedNotification])
+        type = WebExtensionEventListenerType::PermissionsOnAdded;
+    else if ([notificationName isEqualToString:_WKWebExtensionContextGrantedPermissionsWereRemovedNotification])
+        type = WebExtensionEventListenerType::PermissionsOnRemoved;
```

Comment:
Since the function is now `firePermissionsEventListenerIfNecessary`, I think flipping this would look better at the call sites. So `type` is an argument, and `notificationName` is based on what type is passed.